ompi/mca/pml/yalla/pml_yalla.c

```diff
@@ -688,6 +688,7 @@ int mca_pml_yalla_start(size_t count, ompi_request_t** requests)
 
     for (i = 0; i < count; ++i) {
         req = (mca_pml_yalla_base_request_t *)requests[i];
+        sreq = (mca_pml_yalla_send_request_t *)req;
```

Comment:
please cast to sreq/rreq depending th request type, in the `if (req->flags & MCA_PML_YALLA_REQUEST_FLAG_SEND) {` below